src/libraries/System.Private.CoreLib/src/System/Diagnostics/Tracing/EventProvider.cs

```diff
@@ -467,42 +467,59 @@ private unsafe void GetSessionInfo(SessionInfoCallback action, ref List<SessionI
             // does not have this issue.
 #if (TARGET_WINDOWS && (ES_SESSION_INFO || !ES_BUILD_STANDALONE))
             int buffSize = 256;     // An initial guess that probably works most of the time.
-            byte* buffer;
-            while (true)
+            byte* stackSpace = stackalloc byte[buffSize];
```

Comment:
Is the JIT able to constprop `buffSize` here?

Reply:
I assume you mean so that the `stackalloc byte[buffSize]` becomes the equivalent of `stackalloc byte[256]`?  I don't believe so, but it also didn't seem worth the small improvement to introduce a const int that's then used both here and to initialize buffSize.